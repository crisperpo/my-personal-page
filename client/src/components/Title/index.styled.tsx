import styled from 'styled-components';

const TitleContainer = styled.div`
    align-items: center;
    display: flex;
    font-family: 'Bebas Neue', sans-serif;
    flex-flow: column;
    font-size: calc(70px + 9vw);
    font-style: normal;
    font-weight: 400;
    justify-content: center;
    line-height: calc(30px + 8vw);
    opacity: 0.7;
`;

const TitleLine = styled.div`
    display: flex;
    flex-direction: row;
`;

export {
    TitleContainer,
    TitleLine
};
