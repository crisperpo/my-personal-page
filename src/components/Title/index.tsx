import { useContext } from 'react';

import  { AppContext } from '../../context/AppContext';
import Letter from '../Letter';
import { TitleContainer, TitleLine } from './index.styled';

const Title = () => {
    const {pageData} = useContext(AppContext) || {};
    return (
        <TitleContainer>
            {pageData?.title.split(' ').map((titleLine: string, indexLine: number) => (
                <TitleLine key={indexLine}>
                    {titleLine.split('').map((letter: string, indexLetter: number) => (
                        <Letter key={indexLetter} letter={letter} position={ indexLetter + indexLine } />
                    ))}
                </TitleLine>
            ))}
        </TitleContainer>
    );
};

export default Title;
