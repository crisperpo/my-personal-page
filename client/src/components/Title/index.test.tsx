import { render } from '@testing-library/react';

import Title from './index';

describe('Title Component', () => {
    it('renders correctly', () => {
        render(<Title />);
    });
    it('matches the snapshot', () => {
        const { asFragment } = render(<Title />);
        expect(asFragment()).toMatchSnapshot();
    });
});
