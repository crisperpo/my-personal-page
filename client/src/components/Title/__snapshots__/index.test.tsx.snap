// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Title Component > matches the snapshot 1`] = `
<DocumentFragment>
  .c2 {
  color: #E21E25;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c2:hovered {
  animation: emeSZq 3s;
}

.c3 {
  color: #7251A2;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c3:hovered {
  animation: emeSZq 3s;
}

.c4 {
  color: #BED431;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c4:hovered {
  animation: emeSZq 3s;
}

.c5 {
  color: #5ECAE6;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c5:hovered {
  animation: emeSZq 3s;
}

.c6 {
  color: #FE4419;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c6:hovered {
  animation: emeSZq 3s;
}

.c7 {
  color: #F6C715;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c7:hovered {
  animation: emeSZq 3s;
}

.c8 {
  color: #F11AA8;
  mix-blend-mode: color-burn;
  transition: top ease 0.1s;
  transition-delay: 2s;
  position: relative;
}

.c8:hovered {
  animation: emeSZq 3s;
}

.c0 {
  align-items: center;
  display: flex;
  font-family: 'Bebas Neue',sans-serif;
  flex-flow: column;
  font-family: 'Bebas Neue',sans-serif;
  font-size: calc(70px + 9vw);
  font-style: normal;
  font-weight: 400;
  justify-content: center;
  line-height: calc(30px + 8vw);
  opacity: 0.7;
}

.c1 {
  display: flex;
  flex-direction: row;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 letter"
        color="#E21E25"
      >
        C
      </div>
      <div
        class="c3 letter"
        color="#7251A2"
      >
        r
      </div>
      <div
        class="c4 letter"
        color="#BED431"
      >
        i
      </div>
      <div
        class="c5 letter"
        color="#5ECAE6"
      >
        s
      </div>
      <div
        class="c6 letter"
        color="#FE4419"
      >
        t
      </div>
      <div
        class="c7 letter"
        color="#F6C715"
      >
        i
      </div>
      <div
        class="c8 letter"
        color="#F11AA8"
      >
        n
      </div>
      <div
        class="c2 letter"
        color="#E21E25"
      >
        a
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c3 letter"
        color="#7251A2"
      >
        R
      </div>
      <div
        class="c4 letter"
        color="#BED431"
      >
        e
      </div>
      <div
        class="c5 letter"
        color="#5ECAE6"
      >
        s
      </div>
      <div
        class="c6 letter"
        color="#FE4419"
      >
        t
      </div>
      <div
        class="c7 letter"
        color="#F6C715"
      >
        r
      </div>
      <div
        class="c8 letter"
        color="#F11AA8"
      >
        e
      </div>
      <div
        class="c2 letter"
        color="#E21E25"
      >
        p
      </div>
      <div
        class="c3 letter"
        color="#7251A2"
      >
        o
      </div>
    </div>
  </div>
</DocumentFragment>
`;
